<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ScriptWidget</class>
 <widget class="QWidget" name="ScriptWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1115</width>
    <height>788</height>
   </rect>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QToolButton" name="toolButtonInputOutput">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>When enabled, specify initial input / output files used for script iteration (otherwise run script once)</string>
       </property>
       <property name="text">
        <string>Iterate over input / output files</string>
       </property>
       <property name="icon">
        <iconset resource="qchdman.qrc">
         <normaloff>:/images/fileopen.png</normaloff>:/images/fileopen.png</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonTextBesideIcon</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QProgressBar" name="progressBar">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Script status / progress</string>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="format">
        <string>Idle</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="toolButtonRun">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Run this script</string>
       </property>
       <property name="icon">
        <iconset resource="qchdman.qrc">
         <normaloff>:/images/run.png</normaloff>:/images/run.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="toolButtonStop">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Terminate the running script</string>
       </property>
       <property name="icon">
        <iconset resource="qchdman.qrc">
         <normaloff>:/images/stop.png</normaloff>:/images/stop.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QSplitter" name="hSplitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QTableWidget" name="tableWidgetInputOutput">
      <property name="textElideMode">
       <enum>Qt::ElideMiddle</enum>
      </property>
      <property name="showGrid">
       <bool>false</bool>
      </property>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <column>
       <property name="text">
        <string notr="true">$INPUT1$</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string notr="true">$INPUT2$</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string notr="true">$OUTPUT1$</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string notr="true">$OUTPUT2$</string>
       </property>
      </column>
     </widget>
     <widget class="QSplitter" name="vSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QTreeWidget" name="treeWidgetScript">
          <property name="dragDropMode">
           <enum>QAbstractItemView::InternalMove</enum>
          </property>
          <property name="allColumnsShowFocus">
           <bool>true</bool>
          </property>
          <column>
           <property name="text">
            <string>Name</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Type</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Sub-type</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Transition</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Status</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0,0">
          <item row="3" column="2">
           <widget class="QLineEdit" name="lineEditCommand">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Arbitrary command</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QToolButton" name="toolButtonAddCommand">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Add a new command</string>
            </property>
            <property name="text">
             <string>Command</string>
            </property>
            <property name="icon">
             <iconset resource="qchdman.qrc">
              <normaloff>:/images/add-step.png</normaloff>:/images/add-step.png</iconset>
            </property>
            <property name="toolButtonStyle">
             <enum>Qt::ToolButtonTextBesideIcon</enum>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="4">
           <widget class="Line" name="horizontalLine">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QToolButton" name="toolButtonAddProject">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Add a new project</string>
            </property>
            <property name="text">
             <string>Project</string>
            </property>
            <property name="icon">
             <iconset resource="qchdman.qrc">
              <normaloff>:/images/add-step.png</normaloff>:/images/add-step.png</iconset>
            </property>
            <property name="toolButtonStyle">
             <enum>Qt::ToolButtonTextBesideIcon</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QComboBox" name="comboBoxGroupType">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Ignored" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Select group execution mode</string>
            </property>
            <property name="maxVisibleItems">
             <number>20</number>
            </property>
            <item>
             <property name="text">
              <string>Sequential execution</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/sequential-script.png</normaloff>:/images/sequential-script.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Parallel execution</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/parallel-script.png</normaloff>:/images/parallel-script.png</iconset>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QToolButton" name="toolButtonAddGroup">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Add a new group</string>
            </property>
            <property name="text">
             <string>Group</string>
            </property>
            <property name="icon">
             <iconset resource="qchdman.qrc">
              <normaloff>:/images/add-step.png</normaloff>:/images/add-step.png</iconset>
            </property>
            <property name="toolButtonStyle">
             <enum>Qt::ToolButtonTextBesideIcon</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QComboBox" name="comboBoxProjectType">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Ignored" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Select project type</string>
            </property>
            <property name="maxVisibleItems">
             <number>20</number>
            </property>
            <item>
             <property name="text">
              <string>Display information about a CHD input file</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/info.png</normaloff>:/images/info.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Verify the integrity of a CHD input file</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/verify.png</normaloff>:/images/verify.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Copy data from one CHD to another</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/copy.png</normaloff>:/images/copy.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Create a raw CHD from an input file</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/createraw.png</normaloff>:/images/createraw.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Create an HD CHD from an input file</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/createhd.png</normaloff>:/images/createhd.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Create a CD CHD from an input file</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/createcd.png</normaloff>:/images/createcd.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Create an LD CHD from an input file</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/createld.png</normaloff>:/images/createld.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Extract a raw file from a CHD input file</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/extractraw.png</normaloff>:/images/extractraw.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Extract an HD file from a CHD input file</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/extracthd.png</normaloff>:/images/extracthd.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Extract a CD file from a CHD input file</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/extractcd.png</normaloff>:/images/extractcd.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Extract an LD AVI file from a CHD input file</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/extractld.png</normaloff>:/images/extractld.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Dump meta-data from a CHD</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/dumpmeta.png</normaloff>:/images/dumpmeta.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Add meta-data to a CHD</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/addmeta.png</normaloff>:/images/addmeta.png</iconset>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Remove meta-data from a CHD</string>
             </property>
             <property name="icon">
              <iconset resource="qchdman.qrc">
               <normaloff>:/images/delmeta.png</normaloff>:/images/delmeta.png</iconset>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QToolButton" name="toolButtonRemoveGroup">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Remove selected group</string>
            </property>
            <property name="icon">
             <iconset resource="qchdman.qrc">
              <normaloff>:/images/remove-step.png</normaloff>:/images/remove-step.png</iconset>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QToolButton" name="toolButtonRemoveProject">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Remove selected project</string>
            </property>
            <property name="icon">
             <iconset resource="qchdman.qrc">
              <normaloff>:/images/remove-step.png</normaloff>:/images/remove-step.png</iconset>
            </property>
           </widget>
          </item>
          <item row="3" column="3">
           <widget class="QToolButton" name="toolButtonRemoveCommand">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Remove selected command</string>
            </property>
            <property name="icon">
             <iconset resource="qchdman.qrc">
              <normaloff>:/images/remove-step.png</normaloff>:/images/remove-step.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="frameCurrentProject">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>toolButtonInputOutput</tabstop>
  <tabstop>toolButtonRun</tabstop>
  <tabstop>toolButtonStop</tabstop>
  <tabstop>tableWidgetInputOutput</tabstop>
  <tabstop>treeWidgetScript</tabstop>
  <tabstop>toolButtonAddGroup</tabstop>
  <tabstop>comboBoxGroupType</tabstop>
  <tabstop>toolButtonRemoveGroup</tabstop>
  <tabstop>toolButtonAddProject</tabstop>
  <tabstop>comboBoxProjectType</tabstop>
  <tabstop>toolButtonRemoveProject</tabstop>
  <tabstop>toolButtonAddCommand</tabstop>
  <tabstop>lineEditCommand</tabstop>
  <tabstop>toolButtonRemoveCommand</tabstop>
 </tabstops>
 <resources>
  <include location="qchdman.qrc"/>
 </resources>
 <connections/>
</ui>
